scanpy/external/pp/_scrublet.py

```diff
@@ -0,0 +1,487 @@
+    except ImportError:
+        raise ImportError(
+            'Please install scrublet: `pip install scrublet` or `conda install scrublet`.'
+        )
+
+    # Estimate n_neighbors if not provided, and create scrublet object.
+
+    if n_neighbors is None:
+        n_neighbors = int(round(0.5 * np.sqrt(adata_obs.shape[0])))
+
+    scrub = sl.Scrublet(
+        adata_obs.X,
+        n_neighbors=n_neighbors,
+        expected_doublet_rate=expected_doublet_rate,
+        stdev_doublet_rate=stdev_doublet_rate,
+        random_state=random_state,
+    )
+
+    scrub._E_obs_norm = adata_obs.X
+    scrub._E_sim_norm = adata_sim.X
+    scrub.doublet_parents_ = adata_sim.obsm['doublet_parents']
+
+    # Call scrublet-specific preprocessing where specified
+
+    if mean_center and normalize_variance:
+        sl.pipeline_zscore(scrub)
+    elif mean_center:
+        sl.pipeline_mean_center(scrub)
+    elif normalize_variance:
+        sl.pipeline_normalize_variance(scrub)
+
+    # Do PCA. Scrublet fits to the observed matrix and decomposes both observed
+    # and simulated based on that fit, so we'll just let it do its thing rather
+    # than trying to use Scanpy's PCA wrapper of the same functions.
+
+    if mean_center:
+        logg.info('Embedding transcriptomes using PCA...')
+        sl.pipeline_pca(
+            scrub, n_prin_comps=n_prin_comps, random_state=scrub.random_state
+        )
+    else:
+        logg.info('Embedding transcriptomes using Truncated SVD...')
+        sl.pipeline_truncated_svd(
+            scrub, n_prin_comps=n_prin_comps, random_state=scrub.random_state
+        )
+
+    # Score the doublets
+
+    scrub.calculate_doublet_scores(
+        use_approx_neighbors=use_approx_neighbors,
+        distance_metric=knn_dist_metric,
+        get_doublet_neighbor_parents=get_doublet_neighbor_parents,
+    )
+
+    # Actually call doublets
+
+    scrub.call_doublets(verbose=verbose)
+
+    # Store results in AnnData for return
+
+    adata_obs.obs['doublet_score'] = scrub.doublet_scores_obs_
+    adata_obs.obs['predicted_doublet'] = scrub.predicted_doublets_
+
+    # Store doublet Scrublet metadata
+
+    adata_obs.uns['scrublet'] = {}
+    adata_obs.uns['scrublet']['threshold'] = scrub.threshold_
+    adata_obs.uns['scrublet']['doublet_scores_sim'] = scrub.doublet_scores_sim_
+    adata_obs.uns['scrublet']['doublet_parents'] = adata_sim.obsm['doublet_parents']
+    adata_obs.uns['scrublet']['parameters'] = {
+        'expected_doublet_rate': expected_doublet_rate,
+        'sim_doublet_ratio': adata_sim.uns['scrublet']['parameters'][
+            'sim_doublet_ratio'
+        ],
+        'n_neighbors': n_neighbors,
+        'random_state': random_state,
+    }
+
+    if get_doublet_neighbor_parents:
+        adata_obs.uns['scrublet'][
+            'doublet_neighbor_parents'
+        ] = scrub.doublet_neighbor_parents_
+
+    return adata_obs
+
+
+def scrublet_simulate_doublets(
+    adata: AnnData,
+    raw_layer: str = 'raw',
```

Comment:
```suggestion
    layer = None,
```

Then:

```python
X = sc.get._get_obs_rep(adata, layer=layer)
```

This gives you arguments that look like other functions in scanpy, and a default error message.